crates/fuel-core/src/database/state.rs

```diff
@@ -50,37 +59,99 @@ impl StorageMutate<ContractsState> for Database {
         key: &<ContractsState as Mappable>::Key,
         value: &<ContractsState as Mappable>::Value,
     ) -> Result<Option<<ContractsState as Mappable>::OwnedValue>, Self::Error> {
-        Database::insert(self, key.as_ref(), Column::ContractsState, value)
-            .map_err(Into::into)
+        let prev = Database::insert(self, key.as_ref(), Column::ContractsState, value)
+            .map_err(Into::into);
+
+        // Get latest metadata entry
+        let prev_metadata = self
```

Comment:
The same here. You need to modify the specific metadata for the contract, not the last entry in the database=)